lib/cc/yaml/nodes/severity.rb

```diff
@@ -0,0 +1,26 @@
+module CC
+  module Yaml
+    module Nodes
+      class Severity < Scalar
+        SEVERITIS = [
+          "info".freeze,
+          "minor".freeze,
+          "major".freeze,
+          "critical".freeze,
+          "blocker".freeze
+        ].freeze
+
+        def value=(value)
+          unless value.nil?
+            normalized_value = value.to_s.strip.downcase
+            if SEVERITIS.include? normalized_value
+              @value = normalized_value
+            else
+              error(%(Unexpected severity %s), value.inspect)
```

Comment:
It does use format internally.

I used `%()` because at first I quoted the value. Later I switched to `inspect` because technically value can be anything, including `nil` and its string representation is not very useful in the message.